### PR TITLE
Allow new interface style for GraphQL.

### LINKED ISSUE #3600: GraphQL: Support new multiple inheritance syntax
https://github.com/graphql/graphql-js/pull/1169

```graphql
type Foo implements Bar & Baz { field: Type }
```

The above currently fails to parse.

It also replaces this syntax:

```graphql
type Foo implements Bar, Baz { field: Type }
```

We could try to support both by doing:

```js
function parseGraphQL(source) {
  try {
    return parse(source, {allowLegacySDLImplementsInterfaces: false})
  } catch {
    return parse(source, {allowLegacySDLImplementsInterfaces: true})
  }
}
```

### PR DESCRIPTION
**Pull request by ruiaraujo**
This is a breaking change since it will upgrade the old style to the new one.

Closes #3600.

### CODE REVIEW

File: src/language-graphql/parser-graphql.js [hunk 1]
```diff
@@ -34,11 +34,19 @@ function removeTokens(node) {
   return node;
 }
 
+function fallbackParser(parse, source) {
+  try {
+    return parse(source, { allowLegacySDLImplementsInterfaces: false });
+  } catch (_) {
+    return parse(source, { allowLegacySDLImplementsInterfaces: true });
```
Comment (duailibe) [inline src/language-graphql/parser-graphql.js]: It'd be great if we could detect which style the code was originally written so we could support both styles, and we can with this code.

But I don't think it's worth the trouble though.

Comment (ruiaraujo) [reply src/language-graphql/parser-graphql.js]: I would agree with that.

**Comment by suchipi** [comment]
Will users on older versions of GraphQL libraries be inconvenienced if prettier starts using the new style? As in, will their libraries not be able to parse the new `&` style? If so, we should probably go through the effort to detect the style from the source.

**Comment by ruiaraujo** [comment]
Yes, I would consider this change a breaking change and consumers should be made aware.

If you disagree and think that we should support it regardless, I would ask for your suggestions on how to do it since the AST provided does not include any notion of separator.

**Comment by azz** [comment]
We have to check the source (`options.originalText`)

**Comment by ruiaraujo** [comment]
@suchipi @azz I have done it. Feel free to take another look.

**Comment by ruiaraujo** [comment]
Should I worry about the codecov/project check?

**Comment by suchipi** [comment]
Nah

**Pull request merged**